e2e_testing/torchscript/slice_like.py

```diff
@@ -0,0 +1,203 @@
+    @annotate_args([
+        None,
+        ([-1, -1, -1], torch.float32, True),
+    ])
+    def forward(self, x):
+        return x[5:, :3, -1:]
+
+
+@register_test_case(module_factory=lambda: SliceModuleStartEqEnd())
+def SliceModuleStartEqEnd_basic(module, tu: TestUtils):
+    module.forward(tu.rand(6,4,7))
+
+# ==============================================================================
+
+class SliceModuleTwoStep(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1, -1], torch.float32, True),
+    ])
+    def forward(self, x):
+        return x[0:5:2, 0:3:2, 0:4:2]
+
+
+@register_test_case(module_factory=lambda: SliceModuleTwoStep())
+def SliceModuleTwoStep_basic(module, tu: TestUtils):
+    module.forward(tu.rand(10,5,17))
+
+# ==============================================================================
+
+class SliceModuleNegIdx(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.float32, True),
+    ])
+    def forward(self, x):
+        return x[:-1, -2:-1]
+
+
+@register_test_case(module_factory=lambda: SliceModuleNegIdx())
+def SliceModuleNegIdx_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3,9))
+
+# ==============================================================================
+
+class SliceModuleSingleIndex(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.float32, True),
+    ])
+    def forward(self, x):
+        return x[0]
+
+
+@register_test_case(module_factory=lambda: SliceModuleSingleIndex())
+def SliceModuleSingleIndex_basic(module, tu: TestUtils):
+    module.forward(tu.rand(6,8))
+
+# ==============================================================================
+
+class SliceModuleWholeTensor(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.float32, True),
+    ])
+    def forward(self, x):
+        return x[:, :]
+
+
+@register_test_case(module_factory=lambda: SliceModuleWholeTensor())
+def SliceModuleWholeTensor_basic(module, tu: TestUtils):
+    module.forward(tu.rand(6,8))
+
+# ==============================================================================
+
+class SelectIntModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.long, True),
+    ])
+    def forward(self, x):
+        return x.select(0,0)
+
+
+@register_test_case(module_factory=lambda: SelectIntModule())
+def SelectIntModule_basic(module, tu: TestUtils):
+    module.forward(torch.tensor([[0,1,4,5],[3,4,5,6]], dtype=torch.long))
+
+# ==============================================================================
+
+class SelectIntAsSlice(torch.nn.Module):
```

Comment:
Is this same same as `SliceModuleSingleIndex` above? If the intention here is to test different types for the same case, it's better to move the tests close to each other and make it obvious they are just different types. Like `SelectAsSliceInt` vs `SelectAsSliceFloat`.

Reply:
This is actually a case meant to mirror the functionality of a select op, and is just another notation. I wanted to confirm with this test that the select functionality also works when input as a standard slice. 

x[0] == x.select(0, 0)

Reply:
yea, but `x[0] == x.select(0, 0)` already tested by `SliceModuleSingleIndex` test above right?

Reply:
I already know the functionality of those two are the same, I wanted to test that both cases are covered by the e2e path.  If its confusing I'm happy to just remove it

Reply:
> I wanted to test that both cases are covered by the e2e path
yea, but the second case is covered by `SliceModuleSingleIndex` above, right?

Reply:
Lol sorry I edited my comment when I realized exactly what you just said, my bad